server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java

```diff
@@ -110,6 +116,37 @@ public void testRemoteClusterSeedSetting() {
         assertEquals("failed to parse port", e.getMessage());
     }
 
+    public void testRemoteClusterOptionalCredentialSettingValues() {
+        assumeTrue("Skipped test because CCx2 feature flag is not enabled", ClusterSettings.CCX2_FEATURE_FLAG_ENABLED);
+        // simple validation
+        Settings settings = Settings.builder()
+            .put("cluster.remote.foo.authorization", "ApiKey apikey1")
+            .put("cluster.remote.bar.authorization", "ApiKey apikey2")
+            .put("cluster.remote.emptystring.authorization", "")
+            .put("cluster.remote.nullstring.authorization", (String) null)
+            .build();
+        SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION.getAllConcreteSettings(settings).forEach(setting -> setting.get(settings));
+    }
+
+    public void testRemoteClusterAuthorizationSettingDependencies() {
```

Comment:
I wonder if `RemoteClusterSettingsTests` might be a better home for this, given it's testing properties of the remote cluster settings?

Reply:
I moved two new methods `testRemoteClusterEmptyOrNullApiKey` and `testRemoteClusterAuthorizationSettingDependencies` to `RemoteClusterSettingsTests` as you suggested.

I left the pre-existing `testSettingsAreRegistered` in `RemoteClusterServiceTests`.

@gwbrown, do you think it makes sense to move the pre-existing `testSettingsAreRegistered` to `RemoteClusterSettingsTests` too? Or is what I did sufficient to address your feedback?

Reply:
I think leaving the existing test in place is OK. Maybe make a note of it for tech debt week